src/inference/src/dev/core_impl.cpp

```diff
@@ -936,9 +954,21 @@ ov::SoPtr<ov::ICompiledModel> ov::CoreImpl::compile_model(const std::string& mod
         cacheContent.blobId =
             ov::ModelCache::compute_hash(model_str, weights, create_compile_config(plugin, parsed._config));
         std::unique_ptr<CacheGuardEntry> lock = cacheGuard.get_hash_lock(cacheContent.blobId);
+        
+        std::shared_ptr<ov::Model> model = nullptr;
+        bool is_weightless_caching = is_weightless_caching_enabled(parsed._config);
```

Comment:
This is probably not required, 
the function `import_compiled_model`
could add hint (ov::internal::origin_model) property which can be passed to plugin (if plugin support it).

On load_model_from_cache, when loaded model from cache passed to `plugin.import_model` (config can have this property), the plugin logic can check all conditions and read_model using core API if required.  For other devices there can be still same logic